_rules/SC2-5-3-name-from-content-included-in-accessible-name.md

```diff
@@ -0,0 +1,187 @@
+---
+name: Name from content included in accessible name
+description: |
+  Interactive elements labelled through their content must have their visible label as part of their accessible name.
+
+success_criterion:
+- 2.5.3 # Label in Name
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Anne Thyme Nørregaard
+- Bryn Anderson
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+This rule applies to any HTML or SVG element that:
+* has a [semantic role](#semantic-role) that is a [widget role](https://www.w3.org/TR/wai-aria-1.1/#widget_roles) that [supports name from content](https://www.w3.org/TR/wai-aria-1.1/#namefromcontent), and 
+* has [descendants](https://www.w3.org/TR/dom41/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) that are [visible](#visible) [text nodes](https://www.w3.org/TR/dom/#text), and
+* has an [accessible name](#accessible-name), even if the name is the empty string ("").
+
+**Note**: [widget roles](https://www.w3.org/TR/wai-aria-1.1/#widget_roles) that [support name from content](https://www.w3.org/TR/wai-aria-1.1/#namefromcontent) are: `button`, `checkbox`, `gridcell`, `link`, `menuitem`, `menuitemcheckbox`, `menuitemradio`, `option`, `radio`, `searchbox`, `switch`, `tab`, `treeitem`.
+
+**Note:** [Text nodes](https://www.w3.org/TR/dom/#text) with only [whitespace](#whitespace) are not [visible](#visible).
+
+### Expectation
+
+The [visible](#visible) [text nodes](https://www.w3.org/TR/dom/#text) that are [descendants](https://www.w3.org/TR/dom41/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) of the target element are [included](#included-characters) in their entirety within the [accessible name](#accessible-name) of the element, or the [text nodes](https://www.w3.org/TR/dom/#text) do not express anything in [human language](https://www.w3.org/TR/WCAG21/#dfn-human-language-s) and therefore do not live up to the [WCAG definition of text](https://www.w3.org/TR/WCAG21/#dfn-text).
```

Comment:
The success criterion applies to elements with labels that include text or images of text. Thus, an element with a label that does not include text, is inapplicable. I suggest that the text "or the text nodes do not express anything in human language and therefore do not live up to the WCAG definition of text" is removed from the expectation.

Also - add a bullet to the applicability section to show that the rule applies to elements with visible text?

Reply:
@DagfinnRomen, our issue is that the WCAG definition of "text" is ambiguous/subjective and thus cannot be used in the Applicability:

> text: sequence of characters that can be programmatically determined, where the sequence is _expressing something in human language_

(https://www.w3.org/TR/WCAG21/#dfn-text)

Examples where a text node is (probably not in everyone's opinion) expressing something in human language is: 
- X (for "close")
- :-)

Where subjectivity is not allowed in the Applicability, it is allowed in Expectations, so therefore we have had to move the "expresses something in human language" to the Expectation.
Yes, it is a bit weird, that it will then become a "Passed" instead of "Inapplicable" result if the visible label is found to not be expressing something in human language, but we have been back and forth a lot on this, and this is the best option he have been able to come up with.

Reply:
> Also - add a bullet to the applicability section to show that the rule applies to elements with visible text?

We already have this: 

> * has [descendants](https://www.w3.org/TR/dom41/#concept-tree-descendant) in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree) that are [visible](#visible) [text nodes](https://www.w3.org/TR/dom/#text), and

Do you think we need further clarification? 
I guess it would have to be a note then.

Reply:
I think this should have it covered.

Reply:
Thanks for getting back on this, @DagfinnRomen :)